docs: upgrade snaplet + supabase docs. (#15559)
supabase/supabase: apps/docs/pages/guides/integrations/snaplet.mdx

<Admonition type="note">
If you experience any issues with getting Supabase working with Snaplet, or with this guide, feel free to chat with the Snaplet team on [Discord](https://app.snaplet.dev/chat).
</Admonition>

<div>

<img src="/docs/img/guides/integrations/snaplet/snappy-holding-supabase-logo.svg/" alt="Supabase connection string" width="30%" style={{ margin: "auto" }} />

</div>

### Things you'll need before you begin: [#prerequisite]

1. **A production Supabase project’s connection string:** This can be found in Supabase via **Organization > All Projects > Your Project > Project Settings > Connection Pooling > Connection String**. We’ll refer to this in the guide as your “source database” - where your data comes from.

![Supabase connection string](/docs/img/guides/integrations/snaplet/connection_string.webp)

2. A development Supabase project’s connection string: Same steps as above, but a different project/environment. We’ll refer to this in the guide as your “target database” - where we’re restoring the snapshot of the source database to. If you haven’t got this set up yet, we’ll take you through the process of setting this up.

> > **Highly Recommended:** Create a read-only role for your connection string. For more info on how to create a read-only role across all schemas, you can check out the [Snaplet docs](https://docs.snaplet.dev/guides/postgresql#create-a-read-only-role).

Reviewer: Can you convert blockquotes to [admonitions](https://github.com/supabase/supabase/blob/6e1620a20adc22ae3f32ca67a5931f4da4d93344/packages/ui/src/components/Admonition/index.tsx#L5)?

### Step 1: Connect your source database (Production Database) [#connect-source-db]

The first thing you’ll want to do is navigate to [https://www.snaplet.dev/](https://www.snaplet.dev/) and sign up for a new account (it’s free). Once you have successfully signed up for a new account, you’ll begin the onboarding process...

![Onboarding start](/docs/img/guides/integrations/snaplet/onboarding_start.webp)

> On the “Connect database” step, click on "Connect Supabase" to connect your supabase account to snaplet. Proceed through all the steps of authorizing your Supabase organization then selecting Supabase project and giving Snaplet the password associated to your project.

Reviewer: ```suggestion
On the “Connect database” step, click on "Connect Supabase" to connect your Supabase account to snaplet. Authorize your Supabase organization, then select your Supabase project, and enter your project password.
```

![Validating credentials](/docs/img/guides/integrations/snaplet/connect_to_supabase.webp)

### Step 2: Create a snapshot [#create-snapshot]

> Once that is done, continue to the next steps:

Reviewer: ```suggestion
Optionally, you can alter the data in your snapshot:
```

- Subsetting (skipping will create a snapshot with all your data).
- Transforming (skipping will leave all your table columns untouched).

Once you have gone through those steps, a new snapshot process will start and you will now be on the “capture step.” Wait for it to finish - this is the snapshot you will restore into your **target database.**

### Step 3: Create a target database on Supabase (Development Database) [#create-target-db]

Your target database is where you want Snaplet to restore the captured snapshot of your production project. This would most likely be either your staging or development Supabase project.

> If you don’t already have a development database setup on Supabase, you’ll need to create a new database by setting up a new project on Supabase. To create a new project you’ll have to do the following:

Reviewer: ```suggestion
If you don’t already have a development database setup on Supabase, you’ll need to create a new database by setting up a new project on Supabase. To create a new project:
```

1. Go to [app.supabase.io](https://app.supabase.io/)

2. Go to your **organization > All projects**

> 3. In the top left click on your current project name and new project:

Reviewer: ```suggestion
3. In the menu bar, open the project dropdown and click on 'New project`:
```

Avoid using directional descriptors as much as possible, so the docs stay robust if UI shifts around

  ![New project](/docs/img/guides/integrations/snaplet/new_project.webp)

4. Wait for the new database to launch

> Remember the password you use when creating the project. You’ll need this password to connect your database to Snaplet later.

### Step 4: Install the Snaplet CLI [#cli-installation]

1. Open your terminal and run `npm install -g snaplet`
2. Run `snaplet auth login`
3. Navigate to `https://app.snaplet.dev/access-token/cli` to get your access token.
4. Paste the access token in the terminal.

### Step 5: Config setup [#config-setup]

You're now ready to restore your production snapshot into your Supabase development project.

1. Navigate to your project directory

2. Run `snaplet config setup`

>    i. You will be shown a warning to write create `.snaplet/config` in your project directory. Select yes (y) to create the file.

Reviewer: Let's stick with Arabic numerals for all lists, including sublists

   ii. Then you will be asked for a **target database connection string.** These are the database credentials of your **target database that we set up in step 3** (your staging or development database).

   > Note that you need to be using the `postgres` user for your development environment

3. Run `snaplet project setup` - you will be presented with a list of projects, these are databases that are connected to your Snaplet account. Choose the project that contains the snapshot you created in Step 2.

### Step 6: Restore the data target [#restore-data-target]

With all the above steps complete, we can now restore!

1. Run `snaplet snapshot restore --no-reset`

> **Note on `--no-reset`:** This will skip the “reset” step of the restore command. So no existing schemas will be dropped. Learn more about data operations [here](https://docs.snaplet.dev/getting-started/restoring#opting-out)

## All done!

> If you want to learn more about Snaplet, you can explore our docs. If you have any questions, feel free to reach out on [Discord](https://app.snaplet.dev/chat).

Reviewer: This sounds like it was copied over from the Snaplet docs. Can you make it more third-party? 'Our' docs here would refer to Supabase, which I'm guessing is not what you want :D

That’s it! You’re all done and should have restored a version of your Supabase production database with transformed data into your target database. You can now safely code against production-realistic data.

<div>

<img src="/docs/img/guides/integrations/snaplet/snappy-with-supabase-ball.svg/" alt="Supabase connection string" width="30%" style={{ margin: "auto" }} />

</div>

## Supabase local development [#local-development]

> Supabase allows you [develop locally](/docs/guides/getting-started/local-development) through their CLI, you can easily restore any snapshot created in Snaplet to your local instance, just by running:

Reviewer: ```suggestion
Supabase allows you [develop locally](/docs/guides/getting-started/local-development) through the Supabase CLI. You can restore any snapshot created in Snaplet to your local instance by running:
```

```bash
# Login into the Supabase CLI
supabase login

# Initialize Supabase to set up the configuration for developing your project locally:
supabase init

> # With docker installed and running start up your supabase services locally

Reviewer: ```suggestion
# With Docker installed and running, start up your Supabase services locally
```

```bash
Could not drop schema "auth", Snaplet will try to truncate all tables and related objects as a fallback: error: must be owner of schema auth
[Schema] Warning: type "aal_level" already exists, statement: "CREATE TYPE auth.aal_level AS ENUM (...
```

Supabase includes a few schemas that are not owned by the postgres user, for example: **auth, graphql, realtime,** and **storage.** During the capture process, Snaplet will try to capture data for tables under these schemas (if it has permission to read data from them)

Your **target database** may already contain these schemas when restoring. The warnings just mean that when Snaplet attempted to restore these schemas (**auth, graphql,** etc) but was unable to drop the ones already existing in your **target database** (since the user is not an owner of these), and consequently it was not able to create any structure for them (since that structure still exists)

Snaplet will still make sure to clear all data for tables in these schemas and restore data for each of these tables to what is in the snapshot. In other words, **these warnings do not mean that the restore failed, but rather show you what Snaplet tried to do.**

> If you aren't actually needing the data for some of these schemas, you can stop these warnings by excluding the schema from the captured snapshots. You can read more on how to do this [over here in our docs](https://docs.snaplet.dev/references/data-operations/exclude):

Reviewer: ```suggestion
If you don't need the data for some of these schemas, you can stop these warnings by excluding the schema from the captured snapshots. You can read more on how to do this [over here in our docs](https://docs.snaplet.dev/references/data-operations/exclude):
```

    'Get started with Supabase and Snaplet, a developer tool for working with safe, versioned, up-to-date production-like data',
}

This guide will help you get a development Supabase instance set up using Snaplet. In 15 minutes or less, you will create a snapshot of your production Supabase instance and restore it to a development environment you can safely code against.

<Admonition type="note">

> If you experience any issues with getting Supabase working with Snaplet, or with this guide, feel free to come chat with us on [Discord](https://app.snaplet.dev/chat).

Reviewer 2: ```suggestion
If you experience any issues getting Supabase working with Snaplet, feel free to chat with the Snaplet team on [Discord](https://app.snaplet.dev/chat).
```
